### PR TITLE
Feature/asb 891 filter documents by date

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: e2e-tests/cypress/e2e/project-application-documents.feature [hunk 1]
```diff
@@ -89,24 +89,24 @@ Feature: Project Application documents page
...
-			| pre-application        | 2            | Pre-application         | 7   |
-			| developers-application | 7            | Developer's Application | 258 |
-			| acceptance             | 11           | Acceptance              | 266 |
-			| pre-examination        | 6            | Pre-examination         | 24  |
-			| examination            | 20           | Examination             | 671 |
-			| recommendation         | 1            | Recommendation          | 1   |
-			| decision               | 5            | Decision                | 16  |
-			| post-decision          | 1            | Post-decision           | 2   |
+
+	# Scenario Outline: Check <name> filter exist
```
Comment (DavidBeale) [inline e2e-tests/cypress/e2e/project-application-documents.feature:93]: If we are removing this, prob best deleting it

File: e2e-tests/cypress/support/e2e.js [hunk 2]
```diff
@@ -19,3 +19,5 @@ import './commands';
 // Alternatively you can use CommonJS syntax:
 // require('./commands')
 import 'cypress-axe';
+import { slowCypressDown } from 'cypress-slow-down';
+slowCypressDown();
```
Comment (DavidBeale) [inline e2e-tests/cypress/support/e2e.js:23]: Do we need this in the old tests? Best to only add it for the tests which need it.

File: packages/e2e_tests/cypress/support/e2e.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 /// <reference types="@shelex/cypress-allure-plugin" />
 // Import commands.js using ES2015 syntax:
+
 import './commands';
 import '@shelex/cypress-allure-plugin';
 import { slowCypressDown } from 'cypress-slow-down';
 slowCypressDown(); // slows down each command by 500ms
```
Comment (DavidBeale) [inline packages/e2e_tests/cypress/support/e2e.js:7]: Best to only add this for the tests which need it.

File: packages/e2e_tests/cypress/e2e/features/RegistrationPage/5_Registration_footer-links.feature [hunk 4]
```diff
@@ -0,0 +1,20 @@
+Feature: Verify footer links registration page
+	As a Test user
+	I want to click on footer links of the application
+	So that I can verify the functionality
+
+	Background: User start the registration process
+		Given I navigate to the registration start page
+
+	Scenario: click on footer links on the registration page
+		And I click on Terms and conditions footer link
```
Comment (DavidBeale) [inline packages/e2e_tests/cypress/e2e/features/RegistrationPage/5_Registration_footer-links.feature:10]: Still not using any of the abstractions

File: packages/e2e_tests/cypress/e2e/step_definitions/Examination-timeTable/TC_14_Exam-table-footer-link.cy.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import { Then, And } from 'cypress-cucumber-preprocessor/steps';
+
+//Terms and Conditions page
+And('I click on Terms and conditions footer link', () => {
+	cy.clickFooterLink('Terms and conditions');
+});
+
+Then('I am on the Terms and conditions page', () => {
+	cy.pageVerification('Terms and conditions');
+	cy.go('back');
```
Comment (DavidBeale) [inline packages/e2e_tests/cypress/e2e/step_definitions/Examination-timeTable/TC_14_Exam-table-footer-link.cy.js:10]: Surly these back actions should not be part of the validation

File: packages/e2e_tests/cypress/support/customActions/footer-link.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+	clickFooterLink: (linkType) => {
+		switch (linkType) {
+			case 'Sitemap':
+				cy.get('[data-cy="Privacy"]').click();
+				break;
+			case 'Terms and conditions':
+				cy.get('[data-cy="Terms and conditions"]').click();
+				break;
+			case 'Accessibility':
```
Comment (DavidBeale) [inline packages/e2e_tests/cypress/support/customActions/footer-link.js:10]: Would be so easy to make these variables and make it generic

**Review by DavidBeale — CHANGES_REQUESTED** [review]
Really need to make use of abstraction and make the step defs generic IMO

File: e2e-tests/cypress/e2e/project-application-documents.feature (hunk 1 above)
Comment (akano-baba) [reply e2e-tests/cypress/e2e/project-application-documents.feature:93]: Will delete it

File: e2e-tests/cypress/support/e2e.js (hunk 2 above)
Comment (akano-baba) [reply e2e-tests/cypress/support/e2e.js:23]: Will uninstall it

File: packages/e2e_tests/cypress/support/e2e.js (hunk 3 above)
Comment (akano-baba) [reply packages/e2e_tests/cypress/support/e2e.js:7]: Not sure what to do as this happens randomly

File: packages/e2e_tests/cypress/e2e/step_definitions/Examination-timeTable/TC_14_Exam-table-footer-link.cy.js (hunk 5 above)
Comment (akano-baba) [reply packages/e2e_tests/cypress/e2e/step_definitions/Examination-timeTable/TC_14_Exam-table-footer-link.cy.js:10]: Back action is not part of the validation.

File: packages/e2e_tests/cypress/e2e/features/RegistrationPage/5_Registration_footer-links.feature (hunk 4 above)
Comment (akano-baba) [reply packages/e2e_tests/cypress/e2e/features/RegistrationPage/5_Registration_footer-links.feature:10]: Not sure how to use abstractions.

File: packages/e2e_tests/cypress/support/customActions/footer-link.js (hunk 6 above)
Comment (akano-baba) [reply packages/e2e_tests/cypress/support/customActions/footer-link.js:10]: I will look into it

**Comment by akano-baba** [comment]
I will look into making use of abstraction

**Comment by akano-baba** [comment]
@DavidBeale  - Just done a commit
I have addressed some of your comments
slowCypressDown - Uninstalled in old tests
slowCypressDown - Not sure how to address this the new tests
abstractions - Still 
cy.go('back') - removed
Switch case - removed

**Pull request closed without merging**